.github/workflows/test_with_clone.yaml

```diff
@@ -0,0 +1,111 @@
+---
+name: Test Changes with Cloned DB
+
+on:
+    pull_request:
+        types: [ labeled, closed ]
+
+permissions:
+    contents: read
+
+jobs:
+    
+    create_clone_and_run_schemachange:
+        runs-on: ubuntu-latest
+        if: contains(github.event.pull_request.labels.*.name, 'create_clone_and_run_schemachange')
+        environment: dev
+        env:
+            SNOWSQL_PWD: ${{ secrets.SNOWSQL_PWD }}
+            SNOWSQL_ACCOUNT: ${{ secrets.SNOWSQL_ACCOUNT }}
+            SNOWSQL_USER: ${{ secrets.SNOWSQL_USER }}
+            SNOWSQL_WAREHOUSE: ${{ secrets.SNOWSQL_WAREHOUSE }}
+            SNOWSQL_CLONE_ROLE: DATA_ENGINEER
+            SNOWSQL_SCHEMACHANGE_ROLE: SYSADMIN
+            SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE_ORIG: ${{ vars.SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE }}
+            SNOWFLAKE_SYNAPSE_STAGE_STORAGE_INTEGRATION: ${{ vars.SNOWFLAKE_SYNAPSE_STAGE_STORAGE_INTEGRATION }}
+            SNOWFLAKE_SYNAPSE_STAGE_URL: ${{ vars.SNOWFLAKE_SYNAPSE_STAGE_URL }}
+            CLONE_NAME: "${{ vars.SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE }}_${{ github.head_ref }}"
+            STACK: ${{ vars.STACK }}
+
+        steps:
+
+            - uses: actions/checkout@v4
+            - uses: actions/setup-python@v4
+              with:
+                python-version: '3.10'
+
+            - name: Install python libraries
+              shell: bash
+              run: |
+                  pip install schemachange==3.6.1
+                  pip install numpy==1.26.4
+                  pip install pandas==1.5.3
+
+            - name: Install SnowSQL
+              run: |
+                curl -O https://sfc-repo.snowflakecomputing.com/snowsql/bootstrap/1.2/linux_x86_64/snowsql-1.2.9-linux_x86_64.bash
+                SNOWSQL_DEST=~/bin SNOWSQL_LOGIN_SHELL=~/.profile bash snowsql-1.2.9-linux_x86_64.bash
+
+            - name: Verify SnowSQL installation
+              run: ~/bin/snowsql -v
+              
+            - name: Sanitize Clone Name
+              run: |
+                CLONE_NAME_SANITIZED="${CLONE_NAME//[^a-zA-Z0-9_]/_}"
+                echo "Clone name has been updated! The clone name will be: ${CLONE_NAME_SANITIZED}"
+                echo "SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE=${CLONE_NAME_SANITIZED}" >> $GITHUB_ENV
```

Comment:
Nit: This variable name is a little confusing. Without context, I would think that this is _the_ Synapse data warehouse database (i.e., the dev database). Could we make it more explicit that it's a clone by naming it something like `SYNAPSE_DATA_WAREHOUSE_DATABASE_CLONE`?

Reply:
The name convention needs to be this way because of how we run `schemachange`:

* If you look at the `schemachange` command, it points to the `synapse_data_warehouse` folder for where the [config file](https://github.com/Sage-Bionetworks/snowflake/blob/dev/synapse_data_warehouse/schemachange-config.yml) lives.
* This config file pulls the `database_name` from the environment variable `SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE`

In other words, if we change the variable to something else, `schemachange` will report back with an error saying it could not find the `SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE` environment variable.

Reply:
Hmm, yeah this seems to be a potential semantics issue with the way we've set up the schemachange config. To make the config more general that variable really ought to be named something like `TARGET_DATABASE`. But since the config was set up with the assumption that we would only ever be making changes to a specific database (dev/prod Synapse data warehouse) it has a very specific name.

But it is what it is. Not worth refactoring at this point imo.